### PR TITLE
amin implementation

### PR DESCRIPTION
**Pull request by miladm**
Issue [link](https://github.com/pytorch/xla/issues/2967)

### CODE REVIEW

File: test/cpp/test_aten_xla_tensor.cpp [hunk 1]
```diff
@@ -1142,6 +1142,37 @@ TEST_F(AtenXlaTensorTest, TestAllDimKeep) {
   }
 }
 
+TEST_F(AtenXlaTensorTest, TestAMin) {
+  torch::Tensor input =
+      torch::rand({4, 3, 4}, torch::TensorOptions(torch::kFloat));
+  int rank = input.dim();
+  for (bool keepdim : {false, true}) {
+    for (int dim = -rank; dim < rank; ++dim) {
+      auto values_indices = torch::amin(input, {dim}, /*keepdim=*/keepdim);
```
Comment (JackCaoG) [inline test/cpp/test_aten_xla_tensor.cpp:1151]: ditto, what's the return type of `torch::amin`? If it is just `torch::tensor` we should probably spell it.

File: torch_xla/csrc/ops/amin.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  auto lower_for_shape_fn =
+      [&](absl::Span<const xla::XlaOp> operands) -> xla::XlaOp {
+    return BuildMinInDims(operands[0], dimensions, keepdim);
+  };
+  return InferOutputShape({input.shape()}, lower_for_shape_fn);
+}
+
+}  // namespace
+
+AMin::AMin(const Value& input, std::vector<xla::int64> dimensions, bool keepdim)
```
Comment (JackCaoG) [inline torch_xla/csrc/ops/amin.cpp:24]: ditto, `Amin` might be a better name.

File: torch_xla/csrc/reduction.cpp [hunk 3]
```diff
@@ -355,19 +355,25 @@ xla::XlaOp BuildMaxInDims(xla::XlaOp input,
 
 xla::XlaOp BuildMinInDim(xla::XlaOp input, xla::int64 dim,
                          bool keep_reduced_dimensions) {
+  return BuildMaxInDims(input, {dim}, keep_reduced_dimensions);
```
Comment (JackCaoG) [inline torch_xla/csrc/reduction.cpp:358]: I think you meant `BuildMinInDims` here.

File: test/cpp/test_aten_xla_tensor.cpp (hunk 1 above)
Comment (miladm) [reply test/cpp/test_aten_xla_tensor.cpp]: Addressed.

File: torch_xla/csrc/ops/amin.cpp (hunk 2 above)
Comment (miladm) [reply torch_xla/csrc/ops/amin.cpp]: Addressed.

File: torch_xla/csrc/reduction.cpp (hunk 3 above)
Comment (miladm) [reply torch_xla/csrc/reduction.cpp]: Thanks for catching this. Done.

**Pull request closed without merging**